src/Image/Resizer.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2015 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\Image;
+
+use Imagine\Image\Box;
+use Imagine\Image\Point;
+use Imagine\Image\ImagineInterface;
+use Imagine\Gd\Imagine as GdImagine;
+use Contao\CoreBundle\Framework\ContaoFrameworkInterface;
+use Contao\Image as LegacyImage;
+use Contao\File;
+use Contao\System;
+use Contao\Image\Resizer as ImageResizer;
+use Contao\Image\ImageInterface;
+use Contao\Image\ResizeConfigurationInterface;
+use Contao\Image\ResizeCoordinatesInterface;
+use Contao\Image\ResizeOptionsInterface;
+use Contao\Image\ImportantPartInterface;
+use Symfony\Component\Filesystem\Filesystem;
+
+/**
+ * Resizes Image objects via Contao\Image\Resizer and executes legacy hooks.
+ *
+ * @author Martin Auswöger <martin@auswoeger.com>
+ */
+class Resizer extends ImageResizer
+{
+    /**
+     * @var LegacyImage
+     */
+    private $legacyImage;
+
+    /**
+     * @var ContaoFrameworkInterface
+     */
+    private $framework;
+
+    /**
+     * Set the Contao framework
+     */
+    public function setContaoFramework(ContaoFrameworkInterface $framework) {
+        $this->framework = $framework;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function resize(ImageInterface $image, ResizeConfigurationInterface $config, ResizeOptionsInterface $options)
+    {
+        if ((
+            isset($GLOBALS['TL_HOOKS']['executeResize']) &&
+            is_array($GLOBALS['TL_HOOKS']['executeResize']) &&
+            count($GLOBALS['TL_HOOKS']['executeResize'])
+        ) || (
+            isset($GLOBALS['TL_HOOKS']['getImage']) &&
+            is_array($GLOBALS['TL_HOOKS']['getImage']) &&
+            count($GLOBALS['TL_HOOKS']['getImage'])
+        )) {
+
+            @trigger_error(
+                'Using the executeResize and getImage hooks has been deprecated and will no longer work in Contao 5.0. '
+                    . 'Use the resize service instead.',
+                E_USER_DEPRECATED
+            );
+
+            $this->legacyImage = null;
+            $legacyPath = $image->getPath();
+
+            if (strpos($legacyPath, TL_ROOT . '/') === 0 || strpos($legacyPath, TL_ROOT . '\\') === 0) {
```

Comment:
Yes, but we had several problems with exactly this in the core. It depends on where the path comes from, which in this case is `$image->getPath()`.